### PR TITLE
LOGIST-933 Adds option to starting counting from the last commit date

### PR DESCRIPTION
**Pull request by d6o**

### CODE REVIEW

File: github.go [hunk 1]
```diff
@@ -69,12 +69,18 @@ func checkOpenPRs(ctx *context.Context, client *github.Client, owner string, rep
...
+			os.Exit(1)
+		}
+
 		currentPr := prType{
 			client:    client,
 			ctx:       ctx,
 			owner:     owner,
 			repo:      repo,
-			OpenSince: daysSincePRCreated(openPullRequest.CreatedAt),
+			OpenSince: daysSince(initialDate),
```
Comment (boekkooi-fresh) [inline github.go]: This should probably no longer be called `OpenSince`

File: github.go [hunk 2]
```diff
@@ -83,7 +89,29 @@ func checkOpenPRs(ctx *context.Context, client *github.Client, owner string, rep
...
+	var lastDate time.Time
+	for _, commit := range commitsList {
+		lastDate = commit.Commit.Committer.GetDate()
+		fmt.Println(lastDate)
+	}
+
+	return &lastDate, nil
+}
+
+func daysSince(CreatedAt *time.Time) int {
```
Comment (boekkooi-fresh) [inline github.go]: `CreatedAt` is no longer true is it?

File: github.go [hunk 3]
```diff
@@ -83,7 +89,29 @@ func checkOpenPRs(ctx *context.Context, client *github.Client, owner string, rep
...
+func lastCommitDate(ctx *context.Context, client *github.Client, owner string, repo string, pullRequest *github.PullRequest) (*time.Time, error) {
+	commitsList, _, err := client.PullRequests.ListCommits(*ctx, owner, repo, pullRequest.GetNumber(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var lastDate time.Time
+	for _, commit := range commitsList {
+		lastDate = commit.Commit.Committer.GetDate()
+		fmt.Println(lastDate)
```
Comment (boekkooi-fresh) [inline github.go]: Left over debug code?

File: github.go [hunk 4]
```diff
@@ -83,7 +89,29 @@ func checkOpenPRs(ctx *context.Context, client *github.Client, owner string, rep
...
-func daysSincePRCreated(CreatedAt *time.Time) int {
+func initialDate(ctx *context.Context, client *github.Client, owner string, repo string, pullRequest *github.PullRequest) (*time.Time, error) {
+	if !runConfig.DaysSinceLastCommit {
+		return pullRequest.CreatedAt, nil
+	}
+	return lastCommitDate(ctx, client, owner, repo, pullRequest)
+}
+
+func lastCommitDate(ctx *context.Context, client *github.Client, owner string, repo string, pullRequest *github.PullRequest) (*time.Time, error) {
+	commitsList, _, err := client.PullRequests.ListCommits(*ctx, owner, repo, pullRequest.GetNumber(), nil)
```
Comment (boekkooi-fresh) [inline github.go:100]: I'm not sure but maybe [GetUpdatedAt](https://godoc.org/github.com/google/go-github/github#PullRequest.GetUpdatedAt) would be the thing to use?

**Review by boekkooi-fresh — CHANGES_REQUESTED** [review]

File: github.go (hunk 4 above)
Comment (d6o) [reply github.go:100]: GetUpdatedAt is related to everything in the PR, so if someone just edits the PR's description, this field gets updated.

**Review by nenad — APPROVED** [review]

File: github.go [hunk 5]
```diff
@@ -69,27 +69,53 @@ func checkOpenPRs(ctx *context.Context, client *github.Client, owner string, rep
...
+			os.Exit(1)
+		}
+
 		currentPr := prType{
 			client:    client,
 			ctx:       ctx,
 			owner:     owner,
 			repo:      repo,
-			OpenSince: daysSincePRCreated(openPullRequest.CreatedAt),
+			daysSince: daysSince(initialDate),
```
Comment (boekkooi-fresh) [inline github.go]: Maybe `inactiveDays` is a little better then `daysSince` because the first question I have since what?

File: github.go [hunk 6]
```diff
@@ -69,27 +69,53 @@ func checkOpenPRs(ctx *context.Context, client *github.Client, owner string, rep
...
-		fmt.Printf("Repo %s/%s/#%d user '%s' open since '%d' days : ", owner, repo, *openPullRequest.Number, *openPullRequest.User.Login, currentPr.OpenSince)
+		fmt.Printf("Repo %s/%s/#%d user '%s' open since '%d' days : ", owner, repo, *openPullRequest.Number, *openPullRequest.User.Login, currentPr.daysSince)
 		// take action if any
 		actions(currentPr)
 	}
 }
 
-func daysSincePRCreated(CreatedAt *time.Time) int {
-	duration := time.Since(*CreatedAt)
+func initialDate(ctx *context.Context, client *github.Client, owner string, repo string, pullRequest *github.PullRequest) (*time.Time, error) {
```
Comment (boekkooi-fresh) [inline github.go]: maybe lastActionDate or something like that initialDate makes no real sense for a last commit date right?

**Comment by nenad** [comment]
:+1:

**Review by boekkooi-fresh — APPROVED** [review]

**Pull request merged**